skbio/tree/_gme.py

```diff
@@ -0,0 +1,452 @@
+    return root
+
+
+def _average_distance_k(nodek, nodesubtree, dm):
+    """Return the average distance between a subtree, defined
+    by a node and its descendants, and a taxa to attach.
+
+    The subtree here is referred to as a lower subtree.
+
+    """
+    nodelistk = [nodek.name]
+    nodelistsubtree = _tip_or_root(nodesubtree)
+    df = dm.between(nodelistk, nodelistsubtree)
+    return df["value"].mean()
+
+
+def _average_distance_k_upper(nodek, nodesubtree, dm):
+    """Return the average distance between a subtree, defined
+    by all nodes that are not descendants, and a taxa to attach.
+
+    The subtree here is referred to as an upper subtree.
+
+    """
+    nodelistk = [nodek.name]
+    if nodesubtree.is_root():
+        nodelistsubtree = []
+    else:
+        root = nodesubtree.root()
+        nodelistsubtree = [root.name]
+        nodelistsubtree.extend(root.subset() - nodesubtree.subset())
+    df = dm.between(nodelistk, nodelistsubtree)
+    return df["value"].mean()
+
+
+def _lower_subtree_list(ordered, node, dm):
+    """Return the list of values representing the change in tree length
+    after attaching a taxa at an edge not the root edge. The subtree
+    involved is a lower subtree.
+
+    This list is indexed by the tree's postorder.
+
+    """
+    lower_list = []
+    for i, a in enumerate(ordered):
+        if a.is_tip():
+            lower_list.append(dm[a.name, node.name])
+        else:
+            a1, a2 = a.children
+            average = (
+                _subtree_count(a1) * _average_distance_k(node, a1, dm)
+                + _subtree_count(a2) * _average_distance_k(node, a2, dm)
+            ) / _subtree_count(a)
+            lower_list.append(average)
+    return lower_list
+
+
+def _upper_subtree_list(ordered, node, taxa, dm):
+    """Return the list of values representing the change in tree length
+    after attaching a taxa at an edge not the root edge. The subtree
+    involved is an upper subtree.
+
+    This list is indexed by the tree's postorder.
+
+    """
+    upper_list = []
+    for i, a in enumerate(ordered):
+        if a.parent.is_root():
+            upper_list.append(dm[a.parent.name, node.name])
+        else:
+            p = a.parent
+            s_ = a.siblings()
+            for sibling in s_:
+                s = sibling
+            average = (
+                (taxa - _subtree_count(p)) * _average_distance_k_upper(node, p, dm)
+                + _subtree_count(s) * _average_distance_k(node, s, dm)
+            ) / (taxa - _subtree_count(a))
+            upper_list.append(average)
+    return upper_list
+
+
+def _edge_attachment_length(child, lowerlist, upperlist, ordered, taxa, adm):
+    """Return the change in the length of a tree after attaching a new taxa
+    at an edge in comparison to attaching at the root edge.
+
+    The value of attaching at the root edge is considered to be 0.
+
+    """
+    parent = child.parent
+    for node in child.siblings():
```

Comment:
Same here. If you intead to get the last sibling, you can simply do `sibling = child.siblings()[-1]`.